src/main/java/com/google/research/bleth/servlets/ListSimulationsServlet.java

```diff
@@ -34,11 +35,20 @@
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Gson gson = new Gson(); // Used for json serialization.
 
-        // Read SimulationMetadata as a HashMap.
-        ImmutableMap<String, SimulationMetadata> simulations = SimulationMetadata.listSimulations();
+        // Read SimulationMetadata as a sorted LinkedHashMap.
+        Optional<String> sortProperty = Optional.empty();
+        Optional<Query.SortDirection> sortDirection = Optional.empty();
+        if (request.getParameter("sortProperty") != null) {
+            sortProperty = Optional.of(request.getParameter("sortProperty"));
+        }
+        if (request.getParameter("sortDirection") != null) {
+            int sortDirectionIndex = Integer.parseInt(request.getParameter("sortDirection"));
```

Comment:
Why are you passing this as an integer?
Prefer to pass the direction as a string ("ASCENDING" or "DESCENDING"), and use `SortDirection.valueOf(Ascii.toUpperCase(request.getParameter("sortDirection")))`. This way the request URL is easier to understand by us humans (e.g. in logs, when debugging, etc.).